Sources/ToucanSource/Bundle/SourceBundleRenderer.swift

```diff
@@ -143,15 +166,205 @@ public struct SourceBundleRenderer {
+                        || name == input.name
+                    let extMatches =
+                        input.ext == "*" || input.ext.isEmpty
+                        || ext == input.ext
+                    return pathMatches && nameMatches && extMatches
+                }
+            }
+
+            let assetsPath = sourceBundle.config.contents.assets.path
+
+            for content in contents {
+                var assetsReady: Set<String> = .init()
+
+                var behaviors = pipeline.assets.behaviors
+                if behaviors.filter({ $0.id == "copy" }).isEmpty {
+                    behaviors.append(
+                        .init(
+                            id: "copy",
+                            input: .init(
+                                path: "*",
+                                name: "*",
+                                ext: "*"
+                            ),
+                            output: .init(
+                                path: "*",
+                                name: "*",
+                                ext: "*"
+                            )
+                        )
+                    )
+                }
+
+                for behavior in behaviors {
+                    let isAllowed = pipeline.contentTypes.isAllowed(
+                        contentType: content.definition.id
+                    )
+                    guard isAllowed else {
+                        continue
+                    }
+                    let remainingAssets = Set(content.rawValue.assets)
+                        .subtracting(assetsReady)
+                    let inputAssets = filterFilePaths(
+                        from: Array(remainingAssets),
+                        input: behavior.input
+                    )
+
+                    guard !inputAssets.isEmpty else {
+                        continue
+                    }
+
+                    for inputAsset in inputAssets {
+                        let basePath = content.rawValue.origin.path
+                            .split(separator: "/")
+                            .dropLast()
+                            .joined(separator: "/")
+
+                        let sourcePath = [
+                            basePath,
+                            assetsPath,
+                            inputAsset,
+                        ]
+                        .joined(separator: "/")
+
+                        let file = getNameAndExtension(from: inputAsset)
+
+                        let destPath = [
+                            assetsPath,
+                            content.slug.resolveForPath(),
+                            inputAsset,
+                        ]
+                        .joined(separator: "/")
+                        .split(separator: "/")
+                        .dropLast()
+                        .joined(separator: "/")
+
+                        switch behavior.id {
```

Comment:
We should separate each of "behaviour resolvers" into wrappers. This file is complex enough without these. What do you think?